### PR TITLE
feat: support expr for #[prop(default=...)]

### PR DESCRIPTION
**Pull request by ModProg**
- Improved error messages
- Added testing for component macro

fixes: #465

### CODE REVIEW

File: leptos_macro/src/component.rs [hunk 1]
```diff
@@ -407,106 +365,43 @@ impl Docs {
...
-        }
-    }
+#[derive(Clone, Debug, AttributeDerive)]
+#[attribute(ident = prop)]
+struct PropOpt {
+    #[attribute(conflicts = [optional_no_strip, strip_option])]
+    optional: bool,
+    #[attribute(conflicts = [optional, strip_option])]
+    optional_no_strip: bool,
+    #[attribute(conflicts = [optional, optional_no_strip])]
```
Comment (ModProg) [inline leptos_macro/src/component.rs:375]: these are not needed as already implied by the other two fields, just added them for completenes.

**Comment by ModProg** [comment]
Not sure why this fails, and how to fix it

**Comment by gbj** [comment]
Ah bummer yes, looks like we've overwhelmed poor GitHub Actions. I'll run the CI locally later and review.

**Comment by ModProg** [comment]
> Ah bummer yes, looks like we've overwhelmed poor GitHub Actions

Maybe we could split the one test up into multiple actions?

**Comment by gbj** [comment]
Looks great — And thanks for the added tests! (This may well have been what caused the CI to kick the bucket but we'll see.)

**Pull request merged**